client.go

```diff
@@ -90,29 +98,44 @@ func (c *Client) refreshToken() error {
 	return nil
 }
 
+func (c *Client) setState(newState string) {
+	c.lock.Lock()
+	c.state = newState
```

Comment:
Is it necessary to set a variable in a locked environment?

Reply:
Yes, because the variable is accessed across different goroutines. Better ideas welcome ofc.